pkg/controller/admissionchecks/multikueue/workload.go

```diff
@@ -254,7 +257,17 @@ func (a *wlReconciler) reconcileGroup(ctx context.Context, group *wlGroup) error
 				log.V(2).Error(err, "Deleting remote workload", "workerCluster", rem)
 			}
 		}
-		return errors.Join(errs...)
+
+		if !workload.HasQuotaReservation(group.local) && acs.State == kueue.CheckStateRetry {
```

Comment:
So IIUC this code is to clear the admission check status as Pending after the workload is evicted and requeued here: https://github.com/kubernetes-sigs/kueue/blob/7ed44becf8bee6f894cb0e2a6f9cd37c98d7b575/pkg/controller/core/workload_controller.go#L237C30-L246. 

Do we currently keep the state of admission checks on eviction?

Reply:
Only the admission check controller is managing the admission check state.

Reply:
Hm, setting Retry to Pending feels like a job the workload_controller could do on eviction. Is it discussed in KEP?

Reply:
This is not in the scope of this PR. 

`Retry` is the way of keeping a wl for being requeued based on a admission check controller logic.

Reply:
I'm wondering if this is needed in this PR. If we moved the logic to workload_controller, it would not be necessary here. 

@alculquicondor which component should be responsible to transitioning Retry to Pending when a workload is evicted: the individual AC controllers, or this is something the workload_controller could handle?

Reply:
**This is not in the scope of this PR.**

Doing so will prevent other admission check controllers from being able to hold the workload out of the queue.

Reply:
We discussed this offline, and I looked in the KEP, The Retry seems to be associated with `RetryDelayMinutes` which determines the time after which the workload is re-admitted (this could probably be done by workload controller, but TBD). Since we don't support the mechanism yet changing the Retry -> Pending is responsibility of the individual AC controllers. Feel free to resolve.

Reply:
ack